internal/services/springcloud/spring_cloud_gateway_resource.go

```diff
@@ -380,6 +380,96 @@ func resourceSpringCloudGatewayCreateUpdate(d *pluginsdk.ResourceData, meta inte
+	client := meta.(*clients.Client).AppPlatform.GatewayClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	springId, err := parse.SpringCloudServiceID(d.Get("spring_cloud_service_id").(string))
+	if err != nil {
+		return err
+	}
+	id := parse.NewSpringCloudGatewayID(subscriptionId, springId.ResourceGroup, springId.SpringName, d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id.ResourceGroup, id.SpringName, id.GatewayName)
+	if err != nil {
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return fmt.Errorf("checking for existing %s: %+v", id, err)
+		}
+	}
+	if utils.ResponseWasNotFound(existing.Response) {
+		return fmt.Errorf("retreiving %s: resource was not found", id)
+	}
+
+	if existing.Properties == nil {
+		return fmt.Errorf("retreiving %s: properties was nil", id)
+	}
+	properties := existing.Properties
+
+	if existing.Sku == nil {
+		return fmt.Errorf("retreiving %s: sku was nil", id)
+	}
+	sku := existing.Sku
+
+	if d.HasChange("client_authorization") {
+		properties.ClientAuth = expandGatewayClientAuth(d.Get("client_authorization").([]interface{}))
+	}
+
+	if d.HasChange("api_metadata") {
+		properties.APIMetadataProperties = expandGatewayGatewayAPIMetadataProperties(d.Get("api_metadata").([]interface{}))
+	}
+
+	if d.HasChange("application_performance_monitoring_types") {
+		properties.ApmTypes = expandGatewayGatewayApmTypes(d.Get("application_performance_monitoring_types").([]interface{}))
+	}
+
+	if d.HasChange("cors") {
+		properties.CorsProperties = expandGatewayGatewayCorsProperties(d.Get("cors").([]interface{}))
+	}
+
+	if d.HasChange("environment_variables") || d.HasChange("sensitive_environment_variables") {
+		properties.EnvironmentVariables = expandGatewayGatewayEnvironmentVariables(d.Get("environment_variables").(map[string]interface{}), d.Get("sensitive_environment_variables").(map[string]interface{}))
+	}
+
+	if d.HasChange("https_only") {
+		properties.HTTPSOnly = pointer.To(d.Get("https_only").(bool))
+	}
+
+	if d.HasChange("public_network_access_enabled") {
+		properties.Public = pointer.To(d.Get("public_network_access_enabled").(bool))
+	}
+
+	if d.HasChange("quota") {
+		properties.ResourceRequests = expandGatewayGatewayResourceRequests(d.Get("quota").([]interface{}))
+	}
+
+	if d.HasChange("sso") {
+		properties.SsoProperties = expandGatewaySsoProperties(d.Get("sso").([]interface{}))
+	}
+
+	if d.HasChange("instance_count") {
+		sku.Capacity = pointer.To(int32(d.Get("instance_count").(int)))
+	}
+
+	gatewayResource := appplatform.GatewayResource{
+		Properties: properties,
+		Sku:        sku,
+	}
+
+	future, err := client.CreateOrUpdate(ctx, id.ResourceGroup, id.SpringName, id.GatewayName, gatewayResource)
+	if err != nil {
+		return fmt.Errorf("creating/updating %s: %+v", id, err)
+	}
+
+	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
+		return fmt.Errorf("waiting for creation/update of %s: %+v", id, err)
+	}
```

Comment:
Please update the error messages in the create as well 
```suggestion
		return fmt.Errorf("updating %s: %+v", id, err)
	}

	if err := future.WaitForCompletionRef(ctx, client.Client); err != nil {
		return fmt.Errorf("waiting for update of %s: %+v", id, err)
	}
```